src/main/java/eu/dissco/backend/service/AnnotationService.java

```diff
@@ -33,137 +37,90 @@ public class AnnotationService {
-      String path) throws IOException {
-    var annotationsPlusOne = elasticRepository.getLatestAnnotationsJsonResponse(pageNumber,
-        pageSize + 1);
-    return wrapResponse(annotationsPlusOne, pageNumber, pageSize, path);
-  }
-
-  private JsonApiLinksFull buildLinksNode(String path, int pageNumber, int pageSize,
-      boolean hasNextPage) {
-    String pn = "?pageNumber=";
-    String ps = "&pageSize=";
-    String self = path + pn + pageNumber + ps + pageSize;
-    String first = path + pn + "1" + ps + pageSize;
-    String prev = (pageNumber <= 1) ? null : path + pn + (pageNumber - 1) + ps + pageSize;
-
-    String next =
-        (hasNextPage) ? null : path + pn + (pageNumber + 1) + ps + pageSize;
-    return new JsonApiLinksFull(self, first, prev, next);
+    var annotationsPlusOne = repository.getAnnotations(pageNumber, pageSize + 1);
+   return new JsonApiListResponseWrapper(annotationsPlusOne, pageNumber, pageSize, path);
   }
 
-  public AnnotationResponse persistAnnotation(AnnotationRequest annotation, String userId) {
+  public JsonApiWrapper persistAnnotation(AnnotationRequest annotation, String userId, String path) {
     var event = mapAnnotationRequestToEvent(annotation, userId);
     var response = annotationClient.postAnnotation(event);
     if (response != null) {
-      return mapResponseToAnnotationResponse(response);
+      response = response.get("annotation");
+      var type = response.get("type").asText();
+      var id = response.get("id").asText();
+      var dataNode = new JsonApiData(id, type, response);
+      return new JsonApiWrapper(dataNode, new JsonApiLinks(path));
     }
     return null;
   }
 
-  private AnnotationResponse mapResponseToAnnotationResponse(JsonNode response) {
-    var annotation = response.get("annotation");
-    return new AnnotationResponse(response.get("id").asText(),
-        response.get("version").asInt(),
-        annotation.get("type").asText(),
-        annotation.get("motivation").asText(),
-        annotation.get("target"),
-        annotation.get("body"),
-        annotation.get("preferenceScore").asInt(),
-        annotation.get("creator").asText(),
-        Instant.ofEpochSecond(annotation.get("created").asLong()),
-        annotation.get("generator"),
-        Instant.ofEpochSecond(annotation.get("generated").asLong()),
-        null);
+  @NotNull
+  private static AnnotationEvent mapAnnotationRequestToEvent(AnnotationRequest annotation,
+      String userId) {
+    return new AnnotationEvent(annotation.type(), annotation.motivation(), userId, Instant.now(),
+        annotation.target(), annotation.body());
+  }
+
+  public JsonApiWrapper updateAnnotation(String id, AnnotationRequest annotation, String userId, String path)
+      throws NoAnnotationFoundException {
+    var result = repository.getAnnotationForUser(id, userId);
+    if (result > 0) {
+      return persistAnnotation(annotation, userId, path);
```

Comment:
Do we have a user story to change this as we need to support partially filled requests?

Reply:
[Jira](https://naturalis.atlassian.net/browse/DD-390?atlOrigin=eyJpIjoiZGUzYTVhNzI5ZmQxNDRhNWJiMGJhMDczM2E4YjA2NWMiLCJwIjoiaiJ9)